### DISCUSSION TITLE
How to expand LSP hover type definition?

### DISCUSSION THREAD

**Discussion by Raagh** [LazyVim/LazyVim#3842] [Q&A]
I want to expand types that are too big for the hover window. 

![image](https://github.com/LazyVim/LazyVim/assets/8405459/5ab084e1-ad0d-4071-99f1-c0826abfa6a1)

Having the ... doesnt help that much if I want to know the type definition.

**Comment by RHOOPH** [comment]
In your project directory

open  `node_modules/typescript/bin/typescript.js` file

search for `defaultMaximumTruncationLength` 

you should find something like this

```
// src/compiler/utilities.ts
var resolvingEmptyArray = [];
var externalHelpersModuleNameText = "tslib";
var defaultMaximumTruncationLength = 160;
var noTruncationMaximumTruncationLength = 1e6;
```

set the `defaultMaximumTruncationLength` to a higher number

Then restart the lsp
